contrib/go/src/python/pants/contrib/go/tasks/go_fmt.py

```diff
@@ -0,0 +1,44 @@
+# coding=utf-8
+# Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+import subprocess
+
+from pants.base.exceptions import TaskError
+
+from pants.contrib.go.tasks.go_workspace_task import GoWorkspaceTask
+
+
+class GoFmt(GoWorkspaceTask):
+  """Format Go code using gofmt."""
+
+  _GO_SOURCE_EXTENSION = '.go'
+
+  def _is_checked(self, target):
+    return target.has_sources(self._GO_SOURCE_EXTENSION) and not target.is_synthetic
+
+  def execute(self):
+    targets = self.context.targets(self._is_checked)
+    with self.invalidated(targets) as invalidation_check:
+      invalid_targets = [vt.target for vt in invalidation_check.invalid_vts]
+      sources = self.calculate_sources(invalid_targets)
+      if sources:
+        args = [os.path.join(self.go_dist.goroot, 'bin', 'gofmt'), '-w'] + list(sources)
+        try:
+          output = subprocess.check_output(args)
+        except subprocess.CalledProcessError as e:
+          raise TaskError('{} failed with exit code {}'.format(' '.join(args), e.returncode),
+                          exit_code=e.returncode)
+        if output:
+          self.context.logger.info(output)
+
+  def calculate_sources(self, targets):
```

Comment:
I understand the execute body could be a bit different between `GoFmt` and `GoCheckstyle`, but do you think `calculate_sources` and `_is_checked` can be pushed up to `GoWorkspaceTask`?